mantidproject/mantid
# Bugs fixes for Eigen 3.3, use Eigen CMake target

Description of work.

This attempts to use Eigen3's CMake target instead of simply specifying the `EIGEN3_INCLUDE_DIR`. 

Previous test failures appear to be related to the lifetime of temporary `Eigen::Vector3d` objects.  

`Eigen::Quaterniond` is a [Fixed-size vectorizable Eigen object](https://eigen.tuxfamily.org/dox/group__TopicFixedSizeVectorizable.html) and so I believe some precautions need to be taken when using them inside [std::vector](https://eigen.tuxfamily.org/dox/group__TopicStlContainers.html#title2)

**To test:**



* Code Review
* Check that the builds still pass

There is no GitHub issue associated with this pull request

**Release Notes** 

*Does not need to be in the release notes.*

Developers should be notified about support for Eigen v3.3.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
